src/PointwiseFunctions/EquationsOfState/EquationOfState.hpp

```diff
@@ -0,0 +1,215 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace EquationsOfState {
+template <bool IsRelativistic>
+class DarkEnergyFluid;
+template <bool IsRelativistic>
+class IdealFluid;
+template <bool IsRelativistic>
+class PolytropicFluid;
+}  // namespace EquationsOfState
+/// \endcond
+
+/// Contains all equations of state, including base classes
+namespace EquationsOfState {
+/*!
+ * \ingroup EquationsOfStateGroup
+ * \brief Base class for equations of state
+ *
+ * The template parameter `IsRelativistic` is `true` for relativistic equations
+ * of state and `false` for non-relativistic equations of state. The sound speed
+ * and the specific enthalpy have different definitions in the relativistic and
+ * non-relativistic case.
+ */
+template <bool IsRelativistic>
+class EquationOfState : public PUP::able {
+ public:
+  using creatable_classes =
+      tmpl::list<DarkEnergyFluid<IsRelativistic>, IdealFluid<IsRelativistic>,
+                 PolytropicFluid<IsRelativistic>>;
+
+  EquationOfState() = default;
+  EquationOfState(const EquationOfState&) = default;
+  EquationOfState& operator=(const EquationOfState&) = default;
+  EquationOfState(EquationOfState&&) = default;
+  EquationOfState& operator=(EquationOfState&&) = default;
+  ~EquationOfState() override = default;
+
+  WRAPPED_PUPable_abstract(EquationOfState);  // NOLINT
+  virtual std::unique_ptr<EquationOfState<IsRelativistic>> get_clone()
+      const = 0;
+
+  // @{
+  /// Computes the pressure from the rest mass density \f$\rho\f$ and the
+  /// specific internal energy \f$\epsilon\f$
```

Comment:
Personally, I would add a period at end of this sentence (and similar cases below)...